pkg/telemetry/tracing.go

```diff
@@ -220,3 +231,40 @@ func newTextMapPropagator() propagation.TextMapPropagator {
 		propagation.Baggage{},
 	)
 }
+
+// IgnoredSpanProcessor processes spans and does not send them to the next
+// processor if they have the "ignored" attribute set to true.
+//
+// This can be useful for if a span's creation was required for propagation
+// purposes, but turned out not to be used, e.g. if idempotency filtered out a
+// queued item.
+type IgnoredSpanProcessor struct {
+	next trace.SpanProcessor
+}
+
+func NewIgnoredSpanProcessor(next trace.SpanProcessor) *IgnoredSpanProcessor {
```

Comment:
smart.

would there be a case where the ignored span will have child spans, resulting in disruption of the connections between spans because the one in the middle was taken out?

Reply:
Absolutely. The ignored spans atm are always unpropagated spans, so won't have children, but this is brittle, and if a span loses its parent that it'll just use the root.

Right now this is used to remove executions from the trace in order to explicitly not show `StepPlanned` spans as we discussed.